redemptions.go

```diff
@@ -4,40 +4,11 @@ import "encoding/xml"
 
 // Redemption holds redeemed coupons for an account or invoice.
 type Redemption struct {
-	CouponCode             string
-	AccountCode            string
-	SingleUse              NullBool
-	TotalDiscountedInCents int
-	Currency               string
-	State                  string
-	CreatedAt              NullTime
-}
-
-// UnmarshalXML unmarshal a coupon redemption object. Minaly converts href links
-// for coupons and accounts to CouponCode and AccountCodes.
-func (r *Redemption) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
-	var v struct {
-		XMLName                xml.Name   `xml:"redemption"`
-		CouponCode             hrefString `xml:"coupon,omitempty"`
-		AccountCode            hrefString `xml:"account,omitempty"`
-		SingleUse              NullBool   `xml:"single_use,omitempty"`
-		TotalDiscountedInCents int        `xml:"total_discounted_in_cents,omitempty"`
-		Currency               string     `xml:"currency,omitempty"`
-		State                  string     `xml:"state,omitempty"`
-		CreatedAt              NullTime   `xml:"created_at,omitempty"`
-	}
-	if err := d.DecodeElement(&v, &start); err != nil {
-		return err
-	}
-	*r = Redemption{
-		CouponCode:             string(v.CouponCode),
-		AccountCode:            string(v.AccountCode),
-		SingleUse:              v.SingleUse,
-		TotalDiscountedInCents: v.TotalDiscountedInCents,
-		Currency:               v.Currency,
-		State:                  v.State,
-		CreatedAt:              v.CreatedAt,
-	}
-
-	return nil
+	XMLName                xml.Name `xml:"redemption"`
```

Comment:
Based on the example from the latest API docs, the redemption object looks like this: 
```
<redemption href="https://your-subdomain.recurly.com/v2/accounts/1/redemptions/374a1c75374bd81493a3f7425db0a2b8">
  <coupon href="https://your-subdomain.recurly.com/v2/coupons/special" />
  <account href="https://your-subdomain.recurly.com/v2/accounts/1" />
  <subscription href="https://your-subdomain.recurly.com/v2/subscriptions/37bfef7a8e44cfc3817b7a43eba8a6e6">
  <uuid>374a1c75374bd81493a3f7425db0a2b8</uuid>
  <single_use type="boolean">true</single_use>
  <total_discounted_in_cents type="integer">0</total_discounted_in_cents>
  <currency>USD</currency>
  <state>active</state>
  <coupon_code>special</coupon_code>
  <created_at type="datetime">2016-07-11T18:56:20Z</created_at>
  <updated_at type="datetime">2016-07-11T18:56:20Z</updated_at>
</redemption>
```

So the `Redemption` struct needs to look like this: 
```
type Redemption struct {
         UUID                         string  
	CouponCode             string 
	AccountCode            string // hrefString 
        SubscriptionUUID     string // hrefString 
	SingleUse              NullBool
	TotalDiscountedInCents int
	Currency               string
	State                  string
	CreatedAt              NullTime
        UpdatedAt             NullTime 
}
```
You will need to keep the custom unmarshaler. You can find more information on how the `hrefString` type works [here](https://github.com/blacklightcms/recurly/blob/master/type_href.go). Please add an `_Encoding` test for this object type as well.